config.yaml

```diff
@@ -9,6 +9,16 @@ options:
       can be either run in a single container or spread across multiple
       containers, which allows to independently scale each component.
     type: string
+  num-history-shards:
+    default: 0

```

Comment:
Does there need to be a default here, especially as the default is invalid - could you not leave this to default to `None` and then set a `BlockedStatus` to update it? I think a message about needing to add a config value would be more clear than one about an invalid field being set - as the user may not have set any values at this point. 

If you need to ensure it's in the config then you could use a structured config to ensure this and to restrict the values allowed here: [for example](https://github.com/canonical/superset-k8s-operator/blob/main/src/structured_config.py#L174).

Reply:
Good point. I've removed the default value and made changes accordingly.

I should probably move towards using structured config soon, but just haven't found enough justification given the small number of configurations that require validation.

Reply:
One additional benefit of the structured config is that it means that ever config value will be in the config dictionary so it removes the need for `.get()` and validation if it `== ""`. Instead you can just use `if <config_param>`. Something to consider if your parameters with validation continue to grow.